### DISCUSSION TITLE
Hide specific icon in tray module

### DISCUSSION THREAD

**Discussion by liouk** [Alexays/Waybar#2826] [Q&A]
Hi all, is there a way to hide a specific tray icon, maybe with CSS? The application doesn't have an option to disable the tray icon unfortunately.

**Comment by poperigby** [comment]
@liouk did you ever figure this out?

**Reply by liouk** [reply]
Unfortunately, no -- I'm thinking that it should be possible if there was a way to grab the ID or something of the div that encloses the icon, and then just hide that with CSS, but I'm not aware of any way that we can figure out the ID.

**Comment by nihilmd** [comment]
I'm also interested in this, specifically for Spotify, afaik there's no option to disable the tray icon in the application.

I did some digging in the docs and came across the following utility:
` env GTK_DEBUG=interactive waybar -l trace ` 

There does not seem to be a name or label attached to the GtkImage, which probably means there's no way of accessing this element via CSS if I'm understanding this correctly.
The properties are mostly empty as well.

![image](https://github.com/Alexays/Waybar/assets/156137632/debf5d85-96a8-4943-af64-8cc08e23698f)

The trace log for the tray item (probably unrelated but the information is definitely accessible at some point)

![image](https://github.com/Alexays/Waybar/assets/156137632/161d5e68-773b-4133-9efd-c1141f52cb9b)



I might be way off the mark here but maybe setting the name of the item/image before adding it to the EventBox could solve this?
Or wrapping it in a Box with the name if the image does not support this?
`@ src/modules/sni/item.cpp :57 `
